spec/factorial_spec.rb

```diff
@@ -0,0 +1,18 @@
+require_relative '../solver'
+
+# Testing factorial method
+
+describe 'When testing reverse method of Solver Class' do
+  let(:solver) { Solver.new }
+  it 'should return the factorial of number: 5' do
+    expect(solver.factorial(5)).to eq(120)
+  end
+
+  it 'should return the factorial of number: 5' do
+    expect(solver.factorial(10)).to eq(3_628_800)
+  end
+
+  it 'should return the factorial of number: 5' do
+    expect(solver.factorial(8)).to eq(40_320)
+  end
+end
```

Comment:
- You have done great implementing the solver class following the TEST DRIVEN DEVELOPMENT, :+1: But there are a few changes you need to make before moving to the next milestone. :+1: 

> The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.

- [ ] Your `factorial` tests doesn't cover the special case when a negative integer or 0 is passed. I'll suggest you cover these 2 cases in your test and make sure they are implanted in the `factorial` method in the solver class. :+1: :100: :smile: